core/src/main/java/org/elasticsearch/index/Index.java

```diff
@@ -52,6 +52,9 @@
     public Index(String name, String uuid) {
         this.name = Objects.requireNonNull(name).intern();
         this.uuid = Objects.requireNonNull(uuid).intern();
+        if(this.name.contains(":")){
```

Comment:
There should be extra spaces here, between the `if` and the `(`, and the `))` and the `{`.